public/app/plugins/datasource/prometheus/composable_dataquery.cue

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Grafana Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grafanaplugin
+
+import (
+	common "github.com/grafana/grafana/packages/grafana-schema/src/common"
+	"github.com/grafana/grafana/pkg/plugins/pfs"
+)
+
+// This file (with its sibling .cue files) implements pfs.GrafanaPlugin
+pfs.GrafanaPlugin
+
+composableKinds: DataQuery: {
+	maturity: "merged"
+
+	lineage: {
+		seqs: [
+			{
+				schemas: [
+					{
+						common.DataQuery
+						expr: string
+						format?: string
+						instant?: bool
+						range?: bool
+						exemplar?: bool
+						hinting?: bool
+						interval?: string
+						intervalMs?: int64
+						intervalFactor?: int64
+						// Timezone offset to align start & end time on backend
+						utcOffsetSec?: int64
+						legendFormat?: string
+						valueWithRefId?: bool
+						requestId?: string
```

Comment:
what is the thought on where the runtime things should live?  requestId/interval/intervalMs are not valid when saved in panel JSON, but are required/always present when making a request to the backend

Reply:
> are not valid

you're saying that it would be incorrect to include them, because that's ephemeral data, correct?

Yes, what's included in schematized types should ultimately be limited to persisted data, and specifically not include ephemeral data that exists only in the context of a particular request. I'll update our guide accordingly.

There are reasonably straightforward answers here; the main question is whether the additional ephemeral fields are determined by the system, or by the plugin.

Reply:
right -- they are ephemeral and should never be saved as input to queries

The `DataQuery` in the backend has similar properties, but does have a section that is system defined.  These particular properties are added original by the system (PanelQueryRunner) and then the datasource adds the properties to each query

Reply:
yep, PR (and guide docs) are now updated to indicate that only persisted data should be included - an oversight